kratos/containers/pointer_vector_set.h

```diff
@@ -89,11 +89,13 @@ class PointerVectorSet final
     using const_reference = const TDataType&;
     using ContainerType = TContainerType;
 
+    /// Define the iterators
```

Comment:
This comment will be attached to `using iterator ...` in the docs generated by doxygen, and the rest (`const_iterator`, `reverse_iterator`, ...) will be empty, which is quite misleading. I recommend grouping them in a `@name` instead:

```suggestion
    /// @}
    /// @name Iterators
    /// @{
```

\+ matching `@}`